pandas/core/internals/blocks.py

```diff
@@ -2352,7 +2352,7 @@ def _try_coerce_args(self, values, other):
             raise TypeError
         elif is_datetime64_dtype(other):
             # add the tz back
-            other = self._holder(other, dtype=self.dtype)
+            other = self._holder(other.ravel(), dtype=self.dtype)
```

Comment:
How is this change related to the rest of the PR?

Reply:
Here, too, this definitely raised during development without the ravel, but apparently not anymore.  will revert.